pkg/vms/actions.es6

```diff
@@ -0,0 +1,179 @@
+import cockpit from 'base1/cockpit';
+import Libvirt from 'vms/libvirt';
+import { getRefreshInterval } from 'vms/selectors';
+import { VMS_CONFIG } from "vms/config";
+
+/**
+ * All actions dispatchableby in the application
+ */
+
+// --- Provider actions -----------------------------------------
+export function getAllVms () {
+  return virt('GET_ALL_VMS');
+}
+
+export function getVm (lookupId) {
+  return virt('GET_VM', { lookupId }); // provider-specific (i.e. libvirt uses vm_name)
+}
+
+export function shutdownVm (name) {
+  return virt('SHUTDOWN_VM', { name });
+}
+
+export function forceVmOff (name) {
+  return virt('FORCEOFF_VM', { name });
+}
+
+export function rebootVm (name) {
+  return virt('REBOOT_VM', { name });
+}
+
+export function forceRebootVm (name) {
+  return virt('FORCEREBOOT_VM', { name });
+}
+
+export function startVm (name) {
+  return virt('START_VM', { name });
+}
+
+/**
+ * Helper for dispatching virt provider methods.
+ *
+ * Lazily initializes the virt provider and dispatches given method on it.
+ */
+function virt (method, action) {
+  return (dispatch, getState) => getVirtProvider({ dispatch, getState }).then(provider => {
+    if (method in provider) {
+      console.log(`Calling ${provider.name}.${method}(${JSON.stringify(action)})`);
```

Comment:
The debug lines should be hidden behind a toggle switch to turn them on. I'm sure you had plans for that, just figure I'd note it here as a merge req.